tern/analyze/docker/container.py

```diff
@@ -77,23 +77,39 @@ def check_image(image_tag_string):
     logger.debug(
         "Checking if image \"%s\" is available on disk...", image_tag_string)
     try:
-        client.images.get(image_tag_string)
+        image = client.images.get(image_tag_string)
         logger.debug("Image \"%s\" found", image_tag_string)
-        return True
+        return image
     except docker.errors.ImageNotFound:
-        return False
+        return None
 
 
 def pull_image(image_tag_string):
     '''Try to pull an image from Dockerhub'''
     logger.debug("Attempting to pull image \"%s\"", image_tag_string)
     try:
-        client.images.pull(image_tag_string)
+        image = client.images.pull(image_tag_string)
         logger.debug("Image \"%s\" downloaded", image_tag_string)
-        return True
+        return image
     except docker.errors.ImageNotFound:
         logger.warning("No such image: \"%s\"", image_tag_string)
-        return False
+        return None
+
+
+def get_image(image_tag_string):
+    '''Try to get an image from Dockerhub.
+    image_tag_string: can be in image:tag or image@digesttype:digest format'''
+    check_docker_setup()
+    image = check_image(image_tag_string)
+    if image is None:
+        return pull_image(image_tag_string)
+    return image
```

Comment:
`def setup()` in `report.py` has some error checking if neither one of these work:
``` 
if not container.check_image(image_tag_string):
            # if no docker image is present, try to pull it
            if not container.pull_image(image_tag_string):
                logger.fatal("%s", errors.cannot_find_image.format(
                    imagetag=image_tag_string))
                sys.exit()
```
Have you thought about rolling this code in here and using this function in `setup()`?

Reply:
My bad - didn't read your changes to report.py